index.js

```diff
@@ -141,15 +154,16 @@ MapboxLanguage.prototype._initialStyleUpdate = function () {
   this._map.setStyle(this.setLanguage(style, language));
 };
 
-function browserLanguage(supportedLanguages) {
-  const language = navigator.languages ? navigator.languages[0] : (navigator.language || navigator.userLanguage);
-  const parts = language && language.split('-');
-  let languageCode = language;
-  if (parts.length > 1) {
-    languageCode = parts[0];
+function browserLanguage(supportedLanguages, language) {
+  language = language || (navigator.languages ? navigator.languages[0] : (navigator.language || navigator.userLanguage));
+  language = ALT_LOCALES[language.toLowerCase()] || language;
+
+  if (supportedLanguages.indexOf(language) > -1) {
+    return language;
   }
-  if (supportedLanguages.indexOf(languageCode) > -1) {
-    return languageCode;
+  if (language.indexOf('-') > -1) {
+    // reduce longer locales (en-US, zh-Hant-TW) to shorter forms
+    return browserLanguage(supportedLanguages, language.slice(0, language.lastIndexOf('-')));
```

Comment:
This works, though [`reduce()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) would be more succinct, or just a for loop incrementing the end of the range of `parts.join('-')` to look up.

Reply:
I changed these to `while` loops and added tests for "en-US" -> "en" and "zh-HK" to "zh-Hant"
MapBox page showing `name_zh-Hant` is correct label: https://docs.mapbox.com/data/tilesets/reference/mapbox-streets-v8/